File: src/org/cappuccino_project/ide/intellij/plugin/stubs/interfaces/ObjJGlobalVariableDeclarationStub.java
package org.cappuccino_project.ide.intellij.plugin.stubs.interfaces;

import com.intellij.psi.stubs.StubElement;
import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJGlobalVariableDeclarationImpl;
import org.jetbrains.annotations.NotNull;

import javax.annotation.Nullable;

public interface ObjJGlobalVariableDeclarationStub extends StubElement<ObjJGlobalVariableDeclarationImpl> {

    @Nullable
    String getFileName();
    @NotNull
    String getVariableName();
    @Nullable
    String getVariableType();

}

File: src/org/cappuccino_project/ide/intellij/plugin/stubs/interfaces/ObjJGlobalVariableDeclarationStub.kt
package org.cappuccino_project.ide.intellij.plugin.stubs.interfaces

import com.intellij.psi.stubs.StubElement
import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJGlobalVariableDeclarationImpl

interface ObjJGlobalVariableDeclarationStub : StubElement<ObjJGlobalVariableDeclarationImpl> {

    val fileName: String?
    val variableName: String
    val variableType: String?

}
